src/lib/parser/core/PokemonBase.ts

```diff
@@ -444,7 +444,63 @@ export class PokemonBase {
   }
 
   set natureRaw (value: number) {
-    throw new Error(`Setting nature to ${value} directly is not supported. Modify personality instead.`)
+    if (value < 0 || value >= 25) {
+      throw new Error(`Nature value must be between 0 and 24, got ${value}`)
+    }
+
+    if (this.config.setNature) {
+      this.config.setNature(this.data, this.view, value)
+    } else {
+      // Vanilla implementation: modify personality to achieve desired nature
+      // For encrypted Pokemon data, we need to preserve all encrypted data
+      // when changing the personality value, since it affects the encryption key
+
+      // Calculate the current nature and return early if already correct
+      const currentNature = this.personality % 25
+      if (currentNature === value) return
+
+      // First, decrypt all substructs with the current key
+      const substruct0 = this.getDecryptedSubstruct(this.data, 0)
+      const substruct1 = this.getDecryptedSubstruct(this.data, 1)
+      const substruct2 = this.getDecryptedSubstruct(this.data, 2)
+      const substruct3 = this.getDecryptedSubstruct(this.data, 3)
+
+      // Calculate the adjustment needed for the nature
+      const adjustment = value - currentNature
+
+      // Create new personality value that preserves important properties
+      let newPersonality = this.personality + adjustment
+
+      // Handle wraparound cases to ensure we get the exact nature we want
+      const finalNature = newPersonality % 25
+      if (finalNature !== value) {
+        // If we have a negative result or other edge case, adjust accordingly
+        if (finalNature < 0) {
+          newPersonality += 25
+        } else {
+          // Calculate the correction needed
+          const correction = value - finalNature
+          newPersonality += correction
+        }
+      }
+
+      // Ensure the final result gives us the correct nature
+      if (newPersonality % 25 !== value) {
+        // As a last resort, set the personality to a value that definitely works
+        // while trying to preserve as much of the original personality as possible
+        const personalityBase = Math.floor(this.personality / 25) * 25
+        newPersonality = personalityBase + value
+      }
+
+      // Update the personality value in the data
+      this.view.setUint32(this.offsets.personality, newPersonality >>> 0, true)
+
+      // Re-encrypt all substructs with the new key (which uses the new personality)
+      this.setEncryptedSubstruct(0, substruct0)
+      this.setEncryptedSubstruct(1, substruct1)
+      this.setEncryptedSubstruct(2, substruct2)
+      this.setEncryptedSubstruct(3, substruct3)
+    }
   }
```

Comment:
lets also think about if this logic can be simplified and be refactored to be more straight forward